sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3563,6 +3563,186 @@ object functions {
     ToNumber(e.expr, format.expr)
   }
 
+  /**
+   * Returns the ASCII character having the binary equivalent to `n`.
+   * If n is larger than 256 the result is equivalent to char(n % 256)
+   *
+   * @group string_funcs
+   * @since 3.5.0
+   */
+  def char(n: Column): Column = withExpr {
+    Chr(n.expr)
+  }
+
+  /**
+   * Removes the leading and trailing space characters from `str`.
+   *
+   * @group string_funcs
+   * @since 3.5.0
+   */
+  def btrim(str: Column): Column = withExpr {
+    new StringTrimBoth(str.expr)
+  }
+
+  /**
+   * Remove the leading and trailing `trim` characters from `str`.
+   *
+   * @group string_funcs
+   * @since 3.5.0
+   */
+  def btrim(str: Column, trim: Column): Column = withExpr {
+    new StringTrimBoth(str.expr, trim.expr)
+  }
+
+  /**
+   * Returns the character length of string data or number of bytes of binary data.
+   * The length of string data includes the trailing spaces.
+   * The length of binary data includes binary zeros.
+   *
+   * @group string_funcs
+   * @since 3.5.0
+   */
+  def char_length(str: Column): Column = withExpr {
+    Length(str.expr)
+  }
+
+  /**
+   * Returns the character length of string data or number of bytes of binary data.
+   * The length of string data includes the trailing spaces.
+   * The length of binary data includes binary zeros.
+   *
+   * @group string_funcs
+   * @since 3.5.0
+   */
+  def character_length(str: Column): Column = withExpr {
+    Length(str.expr)
+  }
+
+  /**
+   * Returns the ASCII character having the binary equivalent to `n`.
+   * If n is larger than 256 the result is equivalent to chr(n % 256)
+   *
+   * @group string_funcs
+   * @since 3.5.0
+   */
+  def chr(n: Column): Column = withExpr {
+    Chr(n.expr)
+  }
+
+  /**
+   * Returns a boolean. The value is True if right is found inside left.
+   * Returns NULL if either input expression is NULL. Otherwise, returns False.
+   * Both left or right must be of STRING or BINARY type.
+   *
+   * @group string_funcs
+   * @since 3.5.0
+   */
+  def contains(left: Column, right: Column): Column = withExpr {
+    ContainsExpressionBuilder.build("contains", Seq(left.expr, right.expr))
```

Comment:
why not `Contains(left, right)`?

Reply:
Because Contains only supports StringType, not include BinaryType.
https://github.com/apache/spark/blob/f44d9238e40448689477b26dd2ddb4012e5bca5e/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala#L574-L578
https://github.com/apache/spark/blob/f44d9238e40448689477b26dd2ddb4012e5bca5e/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala#L503-L518

Reply:
I see, then I have another question:
does it works with `df.select(contains(col("a"), col("b")))`?
in this case the underlying expressions are not resolved, and I guess it can not get the datatype

Reply:
Yes, that's right, so do we only support stringType here?

Reply:
I think we should only support `StringType` here, and document this limitation (`contains` in sql support both `StringType` and `BinaryType`).

also cc @HyukjinKwon @beliefer @cloud-fan

Reply:
Ok

Reply:
Yeah. I see the limit. Because scala API wrapped column with unresolved expression, it seems we can't reuse the `*ExpressionBuilder` here.